modules/tinymce/src/themes/silver/main/ts/ui/dialog/ImagePreview.ts

```diff
@@ -0,0 +1,164 @@
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { AlloyComponent, Behaviour, Memento, SimpleSpec } from '@ephox/alloy';
+import { Dialog } from '@ephox/bridge';
+import { Cell, Optional } from '@ephox/katamari';
+import { Attribute, Class, Css, Height, Ready, SugarElement, Width } from '@ephox/sugar';
+
+import { ComposingConfigs } from '../alien/ComposingConfigs';
+import { RepresentingConfigs } from '../alien/RepresentingConfigs';
+
+type ImagePreviewSpec = Omit<Dialog.ImagePreview, 'type'>;
+
+export interface ImagePreviewDataSpec {
+  readonly url: string;
+  readonly zoom: Optional<number>;
+  // not documented, but can be helpful when dynamically changing the URL
+  readonly cachedWidth: Optional<number>;
+  readonly cachedHeight: Optional<number>;
+}
+
+const calculateImagePosition = (panelWidth: number, panelHeight: number, imageWidth: number, imageHeight: number, zoom: number) => {
+  const width = imageWidth * zoom;
+  const height = imageHeight * zoom;
+  const left = Math.max(0, panelWidth / 2 - width / 2);
+  const top = Math.max(0, panelHeight / 2 - height / 2);
+
+  return {
+    left: left.toString() + 'px',
+    top: top.toString() + 'px',
+    width: width.toString() + 'px',
+    height: height.toString() + 'px',
+  };
+};
+
+const zoomToFit = (panel: SugarElement<HTMLElement>, width: number, height: number) => {
+  const panelW = Width.get(panel);
+  const panelH = Height.get(panel);
+  return Math.min(panelW / width, panelH / height, 1);
+};
+
+export const renderImagePreview = (spec: ImagePreviewSpec, initialData: Optional<Dialog.ImagePreviewData>): SimpleSpec => {
+  const cachedData = Cell<Dialog.ImagePreviewData>(initialData.getOr({ url: '' }));
+
+  const memImage = Memento.record({
+    dom: {
+      tag: 'img',
+      classes: [ 'tox-imagepreview__image' ],
+      attributes: initialData.map((data) => ({ src: data.url })).getOr({})
+    },
+  });
+
+  const memContainer = Memento.record({
+    dom: {
+      tag: 'div',
+      classes: [ 'tox-imagepreview__container' ],
+      attributes: {
+        role: 'presentation'
+      },
+    },
+    components: [
+      memImage.asSpec()
+    ]
+  });
+
+  const setValue = (frameComponent: AlloyComponent, data: ImagePreviewDataSpec) => {
+    const translatedData: Dialog.ImagePreviewData = {
+      url: data.url
+    };
+    // update properties that are set by the data
+    data.zoom.each((z) => translatedData.zoom = z);
+    data.cachedWidth.each((z) => translatedData.cachedWidth = z);
+    data.cachedHeight.each((z) => translatedData.cachedHeight = z);
+    cachedData.set(translatedData);
+
+    const applyFramePositioning = () => {
+      const imageWidth = translatedData.cachedWidth;
+      const imageHeight = translatedData.cachedHeight;
+
+      if (translatedData.zoom === undefined) {
```

Comment:
Can we please use `Type.isUndefined` here instead as we generally don't like to use `=== undefined`? (it doesn't minify as well and it's a little easier to read)

Reply:
It won't make any difference to minified gzip size, which is the only one that really matters, but sure.